src/record.rs

```diff
@@ -22,41 +24,16 @@ pub trait SrvRecord {
     /// Gets a SRV record's weight.
     fn weight(&self) -> u16;
 
-    /// Parses a SRV record into a URI with a given scheme (e.g. https) and
-    /// `path_and_query` (used as a suffix in the URI).
-    ///
-    /// ```
-    /// # fn srv_record_parse() -> Result<(), http::Error> {
-    /// use srv_rs::{resolver::libresolv::LibResolvSrvRecord, SrvRecord};
-    /// let record = LibResolvSrvRecord {
-    ///     priority: 1,
-    ///     weight: 100,
-    ///     port: 8211,
-    ///     target: String::from("srv-client-rust.deshaw.org"),
-    /// };
-    /// assert_eq!(
-    ///     &record.parse("https", "/")?.to_string(),
-    ///     "https://srv-client-rust.deshaw.org:8211/"
-    /// );
-    /// assert_eq!(
-    ///     &record.parse("http", "/bar")?.to_string(),
-    ///     "http://srv-client-rust.deshaw.org:8211/bar"
-    /// );
-    /// # Ok(())
-    /// # }
-    /// ```
-    fn parse(
-        &self,
-        scheme: impl TryInto<Scheme, Error = impl Into<http::Error>>,
-        path_and_query: impl TryInto<PathAndQuery, Error = impl Into<http::Error>>,
-    ) -> Result<Uri, http::Error> {
-        let scheme: Scheme = scheme.try_into().map_err(Into::into)?;
-        let path_and_query: PathAndQuery = path_and_query.try_into().map_err(Into::into)?;
-        Uri::builder()
-            .scheme(scheme)
-            .path_and_query(path_and_query)
-            .authority(format!("{}:{}", self.target(), self.port()).as_str())
-            .build()
+    /// Parses a SRV record into a URI with a given scheme (e.g. https)
+    fn parse(&self, scheme: Scheme) -> Result<Url, url::ParseError> {
+        let mut url = url::Url::parse("http://h")?;
```

Comment:
pls add a big-ass block comment documenting that this is a bogus value and is only because, for whatever godforsaken reason, there's no way to "build" a `Url` with the `url` crate, so we have to construct a "valid" one, and then `set_foo` the things we care about.